packages/databases-collections/src/modules/create-namespace.ts

```diff
@@ -168,13 +168,18 @@ const reducer: Reducer<CreateNamespaceState, Action> = (
   state = INITIAL_STATE,
   action
 ) => {
-  if (
-    isAction<ResetAction>(action, CreateNamespaceActionTypes.Reset) ||
-    isAction<CloseAction>(action, CreateNamespaceActionTypes.Close)
-  ) {
+  if (isAction<ResetAction>(action, CreateNamespaceActionTypes.Reset)) {
     return { ...INITIAL_STATE };
   }
 
+  if (isAction<CloseAction>(action, CreateNamespaceActionTypes.Close)) {
+    // When a modal is closed, we should not clear the connectionMetaData
+    return {
+      ...INITIAL_STATE,
+      connectionMetaData: state.connectionMetaData,
+    };
+  }
```

Comment:
Drive-by: when opening modal to create a collection after closing it first time, the UI did not show encryption options, because we set `connectionMetaData` to initial state (`{}`).